File: app/src/main/java/rod/bailey/trafficatnsw/json/hazard/HazardProperties.java
package rod.bailey.trafficatnsw.json.hazard;

import android.location.Location;

import org.json.JSONObject;

import java.util.Date;
import java.util.List;

/**
 * GSON domain class - represents a HazardProperties (incident).
 */
public class HazardProperties implements Comparable<HazardProperties> {
    public String adviceA;

    public String adviceB;

    public List<ArrangementElement> arrangementElements;

    public List<String> attendingGroups;

    public long created;

    public String displayName;

    public long end;

    public boolean ended;

    public int hazardId;

    public String headline;

    public boolean impactingNetwork;

    public boolean isInitialReport;

    public boolean isMajor;

    public long lastUpdated;

    public Location latlng;

    public String mainCategory;

    public String otherAdvice;

    public List<Period> periods;

    public JSONObject properties;

    public String publicTransport;

    public List<Road> roads;

    public long start;

    public String subCategoryA;

    public String subCategoryB;

    public String webLinkName;

    public String webLinkUrl;

    /**
     * @return A negative integer if this < another. A positive integer if
     * this > another. 0 if this.equals() another,
     */
    @Override
    public int compareTo(HazardProperties another) {
        int result = 0;

        result = (int) (lastUpdated -  another.lastUpdated);

        return result;
    }

    @Override
    public String toString() {
        StringBuffer buf = new StringBuffer();
        return buf.toString();
    }
}

File: app/src/main/java/rod/bailey/trafficatnsw/json/hazard/HazardProperties.kt
package rod.bailey.trafficatnsw.json.hazard

import android.location.Location

import org.json.JSONObject

import java.util.Date

/**
 * GSON domain class - represents a HazardProperties (incident).
 */
data class HazardProperties(
	var adviceA: String? = null,
	var adviceB: String? = null,
	var arrangementElements: List<ArrangementElement>? = null,
	var attendingGroups: List<String>? = null,
	var created: Long? = 0,
	var displayName: String? = null,
	var end: Long = 0,
	var ended: Boolean = false,
	var hazardId: Int? = 0,
	var headline: String? = null,
	var impactingNetwork: Boolean = false,
	var isInitialReport: Boolean = false,
	var isMajor: Boolean? = false,
	var lastUpdated: Long? = 0,
	var latlng: Location? = null,
	var mainCategory: String? = null,
	var otherAdvice: String? = null,
	var periods: List<Period>? = null,
	var properties: JSONObject? = null,
	var publicTransport: String? = null,
	var roads: List<Road>? = null,
	var start: Long? = 0,
	var subCategoryA: String? = null,
	var subCategoryB: String? = null,
	var webLinkName: String? = null,
	var webLinkUrl: String? = null) : Comparable<HazardProperties> {

	/**
	 * @return A negative integer if this < another. A positive integer if
	 * * this > another. 0 if this.equals() another,
	 */
	override fun compareTo(another: HazardProperties): Int {
		val thisUpdated: Long = lastUpdated ?: 0
		val otherUpdated: Long = another.lastUpdated ?: 0
		return (thisUpdated - otherUpdated).toInt()
	}

}
